index.php

```diff
@@ -13,58 +13,63 @@
  */
-if(php_sapi_name() != 'cli-server') {
-    if(!defined('DOKU_INC')) define('DOKU_INC', dirname(__FILE__).'/');
-    require_once(DOKU_INC.'inc/init.php');
+if (php_sapi_name() != 'cli-server') {
+    if (!defined('DOKU_INC')) define('DOKU_INC', dirname(__FILE__) . '/');
+    require_once(DOKU_INC . 'inc/init.php');
 
-    send_redirect(DOKU_URL.'doku.php');
+    send_redirect(DOKU_URL . 'doku.php');
 }
 
-# ROUTER starts below
+// ROUTER starts below
 
-# avoid path traversal
+// avoid path traversal
 $_SERVER['SCRIPT_NAME'] = str_replace('/../', '/', $_SERVER['SCRIPT_NAME']);
 
-# routing aka. rewriting
-if(preg_match('/^\/_media\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
-    # media dispatcher
+// routing aka. rewriting
+if (preg_match('/^\/_media\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
+    // media dispatcher
     $_GET['media'] = $m[1];
     require $_SERVER['DOCUMENT_ROOT'] . '/lib/exe/fetch.php';
 
-} else if(preg_match('/^\/_detail\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
-    # image detail view
+} elseif (preg_match('/^\/_detail\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
+    // image detail view
     $_GET['media'] = $m[1];
     require $_SERVER['DOCUMENT_ROOT'] . '/lib/exe/detail.php';
 
-} else if(preg_match('/^\/_export\/([^\/]+)\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
-    # exports
+} elseif (preg_match('/^\/_export\/([^\/]+)\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
+    // exports
     $_GET['do'] = 'export_' . $m[1];
     $_GET['id'] = $m[2];
     require $_SERVER['DOCUMENT_ROOT'] . '/doku.php';
 
-} elseif($_SERVER['SCRIPT_NAME'] == '/index.php') {
-    # 404s are automatically mapped to index.php
-    if(isset($_SERVER['PATH_INFO'])) {
-        $_GET['id'] = $_SERVER['PATH_INFO'];
-    }
-    require $_SERVER['DOCUMENT_ROOT'] . '/doku.php';
+} elseif (
+    $_SERVER['SCRIPT_NAME'] !== '/index.php' &&
+    file_exists($_SERVER['DOCUMENT_ROOT'] . $_SERVER['SCRIPT_NAME'])
+) {
+    // existing files
 
-} else if(file_exists($_SERVER['DOCUMENT_ROOT'] . $_SERVER['SCRIPT_NAME'])) {
-    # existing files
-
-    # access limitiations
-    if(preg_match('/\/([\._]ht|README$|VERSION$|COPYING$)/', $_SERVER['SCRIPT_NAME']) or
+    // access limitiations
+    if (preg_match('/\/([._]ht|README$|VERSION$|COPYING$)/', $_SERVER['SCRIPT_NAME']) or
         preg_match('/^\/(data|conf|bin|inc)\//', $_SERVER['SCRIPT_NAME'])
     ) {
         die('Access denied');
     }
 
-    if(substr($_SERVER['SCRIPT_NAME'], -4) == '.php') {
+    if (substr($_SERVER['SCRIPT_NAME'], -4) == '.php') {
         # php scripts
         require $_SERVER['DOCUMENT_ROOT'] . $_SERVER['SCRIPT_NAME'];
     } else {
         # static files
         return false;
     }
+} else {
+    // treat everything else as a potential wiki page
+    // working around https://bugs.php.net/bug.php?id=61286
+    if (isset($_SERVER['PATH_INFO'])) {
+        $_GET['id'] = $_SERVER['PATH_INFO'];
+    } else {
+        $_GET['id'] = $_SERVER['SCRIPT_NAME'];
+    }
```

Comment:
```suggestion
    $request_path = preg_split('/\?/', $_SERVER['REQUEST_URI'], 2)[0];
    if (isset($_SERVER['PATH_INFO'])) {
        $_GET['id'] = $_SERVER['PATH_INFO'];
    } elseif (!($request_path == '/' || $request_path == '/index.php')) {
        $_GET['id'] = $_SERVER['SCRIPT_NAME'];
    }
```

Reply:
All pages with dotted name and non dotted name can be loaded/edited with this one. `http://127.0.0.1/` gives the start page.

I just realize that userewrite is by nature an ambiguous notation. is `/index.php` a scriptname or a pagename ? And, with useslash=1, `/lib/exe/fetch.php` is a scriptname but `/lib/exe/schproutz.php` is a pagename. Funny.

Here, `/index.php` gives the startpage and `?id=index.php` gives the page called index.php. but that's normal.